lib/elixir/lib/process.ex

```diff
@@ -765,6 +765,22 @@ defmodule Process do
     nillify(:erlang.process_info(pid, spec))
   end
 
+  @doc """
+  Returns the stacktrace of the current process.
+
+  To retrieve the stacktrace of an exception use `__STACKTRACE__` instead.
+
+  ## Options
+
+    * `:drop` - Number of stacktrace entries to drop, the call to this
+      function will always add one entry unless it's a tail call
```

Comment:
I am not a fan of the drop option. It can be easily achieved as `Enum.drop`, so I don't buy it. :)

Plus this function has the same tail call issue as `IO.warn`, you need to be aware if you are calling it in the tail notation or not. Fixing `IO.warn` is as easy as adding `:ok` afterwards.

So my vote is to keep only add IO.warn_once as doc false, NOT deprecate `IO.warn` but rather document its limitation, and not add `Process.stacktrace`. I.e. only the first commit.

Reply:
Agree with @josevalim. 

My main request was passing the stacktrace as the third argument of `IO.warn_once/3` instead of the number of stacktrace entries to drop. 

Sorry for the noise @ericmj 🙇

Reply:
@fertapric I don't see the point of that since all callers will have to the same stacktrace dance:

```elixir
{:current_stacktrace, stacktrace} = Process.info(self(), :current_stacktrace)
stacktrace = Enum.drop(stacktrace, ...)
```

Reply:
@ericmj that’s also the case in `IO.warn(message, stacktrace)`

Reply:
Yes, and it often shows incorrect stacktraces that I have fixed in this PR =)

Reply:
Also note that `IO.warn/2` is sometimes used with compile-time traces from the Macro.Env.